### ISSUE TITLE
Add an option to fail startup if plugins fail to load

### ISSUE THREAD

**Issue by Desuuuu** [containous/traefik#10106]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you expect to see?

It would be nice to have the option to have Traefik fail to startup if a plugin fails to load.

Currently, an error is logged and plugins are disabled which can lead to unreachable services if they require plugins to function properly.

The issue has been described in [this forum post](https://community.traefik.io/t/failed-to-download-plugin-caused-all-services-to-go-down/17717) where the user experienced a transient error downloading a plugin, requiring a manual restart to solve the issue.

**Comment by nmengin** [comment]
Hello @Desuuuu,

Thank you for the suggestion.

We are unsure about the use case and the traction it will receive, so we are going to leave the status as "kind/proposal" to give the community time to let us know that they would like this.

We will reevaluate as people respond.

**Comment by ldez** [comment]
> Is there a specific reason behind the current behaviour (logging an error and disabling plugins entirely)?

Yes: the goal is to be able to start Traefik if, for example, our plugin server is down.

> Because as I see it, it renders plugins unusable since any router using any plugin will be unavailable until the issue is fixed and Traefik is restarted.

It's the expected behavior.

This behavior has been added because of a failure of our plugin server: https://github.com/traefik/traefik/pull/9287

**Comment by nmengin** [comment]
Hello there,

We are interested in this feature. To avoid breaking the current behavior, with the other maintainers, we think that makes sense to enable this feature using an option.

If a community member would like to build it, let us know, and we will work with you to ensure you have all the information needed to merge it.
We prefer to work with our community members at the beginning of the design process to ensure that we are aligned and can move quickly with the review and merge process. Let us know here or create a PR before you start, and we will work with you there.

Don’t forget to check out the[ contributor docs](https://github.com/traefik/contributors-guide/blob/master/pr_guidelines.md) and link the PR to this issue.

**Comment by bmagic** [comment]
Hello @nmengin,

I have started something to achieve this (It's only to discuss the design, I haven't tested anything).
https://github.com/bmagic/traefik/commit/28739cbd7801071b81c917c96ac639aa2af9110b

So I've added a `Required` field in the plugin descriptor and for each plugin I'll check if it's required.   
On the plugin client, I'll return an error if the plugin is required and something serious is going on, or I'll only return a warning if the plugin is not required. 

This implementation changes the default behavior, because currently if a single plugin has a problem during download/check/unzip, no plugin is instantiated. With this change, all plugins that have been successfully download/check/unzip will be activated.
